### ISSUE TITLE
[MPS] linear with in_features=0 aborts the process; backward also fails

### ISSUE THREAD

**Issue by bogatyy** [pytorch/pytorch#190050]
### 🐛 Describe the bug

`F.linear` with `in_features = 0` hard-crashes the whole process with SIGABRT
on macOS 15+. CPU (and CUDA) return the broadcast bias, or zeros without bias.
The MPS forward only early-returns when the *output* is empty, which it is not
for `K = 0, N > 0`, and then builds an MPSNDArray over a zero-length dimension,
which trips an uncatchable MetalPerformanceShaders assertion.

```python
import torch, torch.nn.functional as F

x = torch.randn(3, 4, 0)
w = torch.randn(5, 0)
b = torch.randn(5)
print(F.linear(x, w, b).shape)  # CPU: (3, 4, 5), equals broadcast bias

out = F.linear(x.to("mps"), w.to("mps"), b.to("mps"))  # SIGABRT
```

Output on a source build of current main (2.14.0a0+git2662808); also
reproduces on 2.13.0 stable:

```text
torch.Size([3, 4, 5])
MPSNDArray.mm:759: failed assertion `[MPSNDArray initWithDevice:descriptor:isTextureBacked:] Error: device may not be nil'
[process aborts, rc=-6]
```

`nn.Linear(0, 5)` crashes the same way; bias is not required. `torch.matmul`
and `torch.addmm` with `K = 0` on MPS are fine — only linear's macOS-15+
no-graph fast path is unguarded (the pre-macOS-15 MPSGraph path fails loudly
with a catchable "Placeholder tensor is empty!" instead).

Backward is independently broken: the input-gradient path checks
`grad_output.numel()` rather than the (empty) grad-input, and the weight path
reshapes and runs a graph for the empty `(N, 0)` weight gradient, so direct
`aten.linear_backward` calls for input, weight, and bias masks all raise.
CPU returns valid empty gradients and `grad_bias = grad_output.sum(...)`.

The forward guard is at
https://github.com/pytorch/pytorch/blob/26628087023e5102381849e9a27c3f09b6d485a4/aten/src/ATen/native/mps/operations/Linear.mm#L119-L121
and the backward guards at
https://github.com/pytorch/pytorch/blob/26628087023e5102381849e9a27c3f09b6d485a4/aten/src/ATen/native/mps/operations/Linear.mm#L238-L240
and
https://github.com/pytorch/pytorch/blob/26628087023e5102381849e9a27c3f09b6d485a4/aten/src/ATen/native/mps/operations/Linear.mm#L311-L315
check only `grad_output`.

### Versions

PyTorch 2.14.0a0+git2662808 (source build of main) and 2.13.0; macOS 26.5.2 (arm64), Apple M5 Max, Python 3.12.13.

cc @ezyang @gchanan @kadeng @msaroufim @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20
